### ISSUE TITLE
Problem with crawling 5.4

### ISSUE THREAD

**Issue by gurudlan** [spatie/laravel-sitemap#88]
Hello,

I have lara app 5.4v and your package 2.4.0. I create sitemap.xml via command
 _$path = public_path(). '/sitemap.xml';
 $SiteMap = SitemapGenerator::create('https://cms.localhost')->writeToFile($path);_
but, xml look's like:
_<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
    <url>
        <loc>https://cms.localhost/</loc>

        <lastmod>2017-09-05T20:12:21+00:00</lastmod>
    
        <changefreq>daily</changefreq>
    
        <priority>0.8</priority>
    </url>
</urlset>_

1.problem is big space between tags...
2.problem is, that generate online web base loc(homepage...). on my site is sub page like as contact ...and sitematgenerator generate xml only with base path...
3.problem is, that automatically set priority to 0.8, but on base path(home page = primary), i need set priority to 1...

Can you help me? 

Thanks

**Comment by uyab** [comment]
@gurudlan are you sure your local URL are `cms.localhost` ? If not, you can change it in file `.env`.

**Comment by gurudlan** [comment]
@uyab  yes i'm sure...in .env + config/app.php i have https://cms.localhost...I try clear all cache, optimize, config clear...But problem still, generate only base path without sub-pages... Any idea?

**Comment by gurudlan** [comment]
But, i have multylanguage site and URL is: https://cms.localhost/sk (base path for SK) and https://cms.localhost/en (base path for EN)...and sub-pages are on https://cms.localhost/en/contact ...

**Comment by Lloople** [comment]
I'm having the same problem in 5.5. The crawler generated all the urls linked in the home but skipped the langmenu on the top of the webpage. Any idea?

**Comment by gurudlan** [comment]
Hello,

i fix problem 1 and 3.
for fix problem 1 with big space a change blade:

`<url>
@if (! empty($tag->url))
        <loc>{{ $tag->url }}</loc>
@endif
@if (count($tag->alternates))
@foreach ($tag->alternates as $alternate)
            <xhtml:link rel="alternate" hreflang="{{ $alternate->locale }}" href="{{ $alternate->url }}" />
@endforeach
@endif
@if (! empty($tag->lastModificationDate))
        <lastmod>{{ $tag->lastModificationDate->format(DateTime::ATOM) }}</lastmod>
@endif
@if (! empty($tag->changeFrequency))
        <changefreq>{{ $tag->changeFrequency }}</changefreq>
@endif
@if (! empty($tag->priority))
        <priority>{{ $tag->priority }}</priority>
@endif
    </url>
`

and for fix problem 3 with auto set priority 1 to home-page a change default value of variable in "url.php:
`    /** @var float */
    public $priority = 1;`

but problem 2 with no-crawler sub-pages still...Any idea for this problem?

Thanks

**Comment by Lloople** [comment]
@gurudlan I gave you the answer on my previous reply. Check it out https://github.com/spatie/laravel-sitemap/issues/88#issuecomment-334459361

**Comment by gurudlan** [comment]
@Lloople  and how you execute with JavaScript?
